src/Spectre.libHeatmapDataScaling/SuppressionAlgorithm.cpp

```diff
@@ -0,0 +1,51 @@
+/*
+* SupressionAlgorithm.cpp
+* Class with suppresion algorithm implementation for contrast enhancement.
+*
+Copyright 2018 Daniel Babiak
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "Spectre.libFunctional/Transform.h"
+#include "Spectre.libHeatmapDataScaling/SuppressionAlgorithm.h"
+
+namespace spectre::visualization
+{
+SuppressionAlgorithm::SuppressionAlgorithm(const double topPercent) : m_topPercent(topPercent) { }
+
+std::vector<double> SuppressionAlgorithm::scaleData(const gsl::span<const double> intensities)
+{
+    const double maxIntensity = *max_element(std::begin(intensities), std::end(intensities));
+    const double cutoff = calculateQuantile(intensities, 1 - m_topPercent);
```

Comment:
If this `1 - m_topPercent` passes, you have *warning level* set to 3 instead of 4, or disabled *treating warnings as errors*.